frontend/src/pages/TaskListPage.tsx

```diff
@@ -1,9 +1,144 @@
+import { modalsActions } from '../redux/modals/index';
+import { ModalTypes } from '../redux/modals/types';
+import { Roadmap } from '../redux/roadmaps/types';
+import { RootState } from '../redux/types';
+import { userInfoSelector } from '../redux/user/selectors';
+import { UserInfo } from '../redux/user/types';
+import { RoleType } from '../../../shared/types/customTypes';
+import { FilterTypes } from '../utils/TaskUtils';
+import { titleCase } from '../utils/string';
+import css from './TaskListPage.module.scss';
+import {
+  chosenRoadmapSelector,
+  allTasksSelector,
+} from '../redux/roadmaps/selectors';
+import { getType } from '../utils/UserUtils';
+
+const classes = classNames.bind(css);
 
 export const TaskListPage = () => {
+  const { t } = useTranslation();
   const tasks = useSelector(allTasksSelector(), shallowEqual);
-  return <TaskTable tasks={tasks} />;
+  const [checked, setChecked] = useState(true);
+  const [searchString, setSearchString] = useState('');
+  const [searchFilter, setSearchFilter] = useState(FilterTypes.SHOW_ALL);
+
+  const userInfo = useSelector<RootState, UserInfo | undefined>(
+    userInfoSelector,
+    shallowEqual,
+  );
+  const currentRoadmap = useSelector<RootState, Roadmap | undefined>(
+    chosenRoadmapSelector,
+    shallowEqual,
+  );
+
+  const dispatch = useDispatch<StoreDispatchType>();
+
+  const onSearchChange = (value: string) => {
+    setSearchString(value.toLowerCase());
+  };
+
+  const toggleCheckedClicked = () => {
+    setChecked(!checked);
+    if (checked) setSearchFilter(FilterTypes.NOT_RATED_BY_ME);
+    else setSearchFilter(FilterTypes.SHOW_ALL);
+  };
+
+  const onAddNewTaskClick = (e: any) => {
+    e.preventDefault();
+    dispatch(
+      modalsActions.showModal({
+        modalType: ModalTypes.ADD_TASK_MODAL,
+        modalProps: {},
+      }),
+    );
+  };
+
+  const onImportTasksClick = (name: string) => (e: any) => {
+    e.preventDefault();
+    dispatch(
+      modalsActions.showModal({
+        modalType: ModalTypes.IMPORT_TASKS_MODAL,
+        modalProps: { name },
+      }),
+    );
+  };
+
+  const renderImportButton = (name: string) => {
+    // TODO: disable button if oauth is not completed
+    // or maybe open the oauth modal first then
+    if (currentRoadmap?.integrations.some((it) => it.name === name)) {
+      return (
+        <button
+          className={classes(css['button-small-filled'])}
+          type="submit"
+          onClick={onImportTasksClick(name)}
+        >
+          <Trans i18nKey="Import tasks from" /> {titleCase(name)}
+        </button>
+      );
+    }
+    return null;
+  };
+
+  const renderTopbar = () => {
+    return (
+      <div className={classes(css.topBar)}>
+        <div className={classes(css.searchBarContainer)}>
+          <>
+            <input
+              className={classes(css.search)}
+              placeholder={t('Search for tasks')}
+              onChange={(e: any) => onSearchChange(e.currentTarget.value)}
```

Comment:
```suggestion
              onChange={(e) => onSearchChange(e.currentTarget.value)}
```

Type inference should work fine